CRM/Contact/Form/Task/SaveSearch.php

```diff
@@ -232,6 +236,20 @@ public function postProcess() {
 
     CRM_Contact_BAO_Group::create($params);
 
+    // Update mapping with the name and description of the group.
+    if ($mappingId) {
+      $mappingParams = array(
+        'id' => $mappingId,
+        'name' => CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_Group', $group->id, 'name', 'id'),
```

Comment:
I had it first that way, but thought it was nicer to have the group name instead of the group title from the submitted value there since the column name suggests having the name.
Your fatal means, thats creating / saving the smart group actually failed? You got any instructions how to recreate that?
`if ($mappingId && $group) {`
should be sufficient then?